### ISSUE TITLE
Test failures when target-cpu is set

### ISSUE THREAD

**Issue by unzvfu** [mir-protocol/plonky2#675]
I consistently see particular test failures when, and only when, `target-cpu` is set:
```
~/src/plonky2$ rustup show
[...]
active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.65.0-nightly (86c6ebee8 2022-08-16)

~/src/plonky2$ git status
On branch main
Your branch is up to date with 'origin/main'.
~/src/plonky2$ cargo clean
[...]
~/src/plonky2$ RUSTFLAGS='-C target-cpu=native' cargo test --release
[...]
test memory::memory_stark::tests::test_stark_circuit ... ok
test cpu::kernel::tests::curve_ops::bn::test_ec_ops ... ok
test cpu::kernel::tests::curve_ops::secp::test_ec_ops ... ok
test all_stark::tests::test_all_stark_recursive_verifier ... FAILED
test all_stark::tests::test_all_stark ... FAILED
test cpu::cpu_stark::tests::test_stark_circuit ... ok
test logic::tests::test_stark_circuit ... ok
test cpu::kernel::tests::ecrecover::test_ecrecover ... ok
test keccak::keccak_stark::tests::test_stark_circuit ... ok

failures:

---- all_stark::tests::test_all_stark_recursive_verifier stdout ----
thread '<unnamed>' panicked at 'Quotient has failed, the vanishing polynomial is not divisible by Z_H: Condition failed: `self.len() >= len`', evm/src/prover.rs:255:18

[...two lines above are repeated three more times...]


failures:
    all_stark::tests::test_all_stark
    all_stark::tests::test_all_stark_recursive_verifier

test result: FAILED. 69 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.77s

error: test failed, to rerun pass '-p plonky2_evm --lib'
```
When running `cargo test --release` without setting `RUSTFLAGS`, the error disappears.

I've done several iterations of `cargo clean; RUSTFLAGS='...' cargo test --release` and the error is always the same.

I think the error first started appearing after a `rustup update` (can't remember which one though, unfortunately), which is what makes me think/hope it is a compiler problem.

My compiler config:
```
$ rustc -Ctarget-cpu=native --print cfg
debug_assertions
panic="unwind"
target_abi=""
target_arch="x86_64"
target_endian="little"
target_env="gnu"
target_family="unix"
target_feature="adx"
target_feature="aes"
target_feature="avx"
target_feature="avx2"
target_feature="avx512bitalg"
target_feature="avx512bw"
target_feature="avx512cd"
target_feature="avx512dq"
target_feature="avx512f"
target_feature="avx512gfni"
target_feature="avx512ifma"
target_feature="avx512vaes"
target_feature="avx512vbmi"
target_feature="avx512vbmi2"
target_feature="avx512vl"
target_feature="avx512vnni"
target_feature="avx512vp2intersect"
target_feature="avx512vpclmulqdq"
target_feature="avx512vpopcntdq"
target_feature="bmi1"
target_feature="bmi2"
target_feature="cmpxchg16b"
target_feature="ermsb"
target_feature="f16c"
target_feature="fma"
target_feature="fxsr"
target_feature="llvm14-builtins-abi"
target_feature="lzcnt"
target_feature="movbe"
target_feature="pclmulqdq"
target_feature="popcnt"
target_feature="rdrand"
target_feature="rdseed"
target_feature="sha"
target_feature="sse"
target_feature="sse2"
target_feature="sse3"
target_feature="sse4.1"
target_feature="sse4.2"
target_feature="ssse3"
target_feature="xsave"
target_feature="xsavec"
target_feature="xsaveopt"
target_feature="xsaves"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_has_atomic_equal_alignment="16"
target_has_atomic_equal_alignment="32"
target_has_atomic_equal_alignment="64"
target_has_atomic_equal_alignment="8"
target_has_atomic_equal_alignment="ptr"
target_has_atomic_load_store="16"
target_has_atomic_load_store="32"
target_has_atomic_load_store="64"
target_has_atomic_load_store="8"
target_has_atomic_load_store="ptr"
target_os="linux"
target_pointer_width="64"
target_thread_local
target_vendor="unknown"
unix
```

**Comment by Sladuca** [comment]
I also just noticed this happening!

**Comment by Sladuca** [comment]
So I actually haven't been using that flag until I saw this. I'm getting this on ` rustc 1.64.0-nightly` as well as `rustc 1.63.0-nightly` too on x86.

```
$ rustc -Ctarget-cpu=native --print cfg
debug_assertions
panic="unwind"
target_abi=""
target_arch="x86_64"
target_endian="little"
target_env="gnu"
target_family="unix"
target_feature="adx"
target_feature="aes"
target_feature="avx"
target_feature="avx2"
target_feature="avx512bw"
target_feature="avx512cd"
target_feature="avx512dq"
target_feature="avx512f"
target_feature="avx512vl"
target_feature="avx512vnni"
target_feature="bmi1"
target_feature="bmi2"
target_feature="cmpxchg16b"
target_feature="ermsb"
target_feature="f16c"
target_feature="fma"
target_feature="fxsr"
target_feature="llvm14-builtins-abi"
target_feature="lzcnt"
target_feature="movbe"
target_feature="pclmulqdq"
target_feature="popcnt"
target_feature="rdrand"
target_feature="rdseed"
target_feature="rtm"
target_feature="sse"
target_feature="sse2"
target_feature="sse3"
target_feature="sse4.1"
target_feature="sse4.2"
target_feature="ssse3"
target_feature="xsave"
target_feature="xsavec"
target_feature="xsaveopt"
target_feature="xsaves"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_has_atomic_equal_alignment="16"
target_has_atomic_equal_alignment="32"
target_has_atomic_equal_alignment="64"
target_has_atomic_equal_alignment="8"
target_has_atomic_equal_alignment="ptr"
target_has_atomic_load_store="16"
target_has_atomic_load_store="32"
target_has_atomic_load_store="64"
target_has_atomic_load_store="8"
target_has_atomic_load_store="ptr"
target_os="linux"
target_pointer_width="64"
target_thread_local
target_vendor="unknown"
unix
```

**Comment by Sladuca** [comment]
So what appears to be happening is is that the each `quotient_poly` in `quotient_polys` returned by `compute_quotient_polys` has a length 1/4 of what it should be.

`compute_quotient_polys` is [evaluating `P::WIDTH` elements at a time](https://github.com/mir-protocol/plonky2/blob/cbfc13c33f368f1051607f142251c1263621f1cc/starky/src/prover.rs#L260), which is 1 without AVX but 4 with AVX. But in [`constraint_consumer.accumulators`](https://github.com/mir-protocol/plonky2/blob/cbfc13c33f368f1051607f142251c1263621f1cc/starky/src/constraint_consumer.rs#L48) we're only returning a single element of each accumulator's packed field, which works fine if there's `P::WIDTH` is 1 but not if it's 4.

Is this behavior intentional or should it be accumulating all four elements of the `PackedField` into the quotient polynomial?

**Comment by Sladuca** [comment]
It turns out `constraint_consumer` is only emitting 1 element of the packed field, so we're losing the other 3 elements when using `Avx2GoldilocksField`. Put up a fix in #675
